velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1399,4 +1400,15 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct ToISO8601FromDateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Date>& date) {
+    result = DATE()->toString(date);
```

Comment:
toString() methods are meant for logging and debugging. The output is not stable and therefore cannot be used like that. Let's add a static method toISO8601(int32_t days) somewhere and use that.

Reply:
I feel adding such a function would end up being a rehashing of `Timestamp::tmToString()`, the logic to construct the output string would be redundant, wouldn't it?

Reply:
@svm1 I'm not sure I understand this. Would you clarify?

Reply:
@mbasmanova Sure, I just didn't quite understand why you say the output of toString() is not stable. A static method added explicitly for this purpose would inevitably use similar logic to construct the output string (as it currently exists in `DATE()->toString()`). A logical alternative would be to use the external date library API for the conversion, but that would just be relying on an external stringify rather than our own.

Reply:
@svm1 By convention, toString() methods are for use in logging and debugging and their output is not guaranteed to be stable. They often don't even document the output format. On the contrary, a specific toISO8601 method would document the output format and stick to it. The toString() method may be implemented by calling toISO8601() method to avoid copy-paste. Notice that you modified the output of toString() method without checking whether some caller is relying on the old format of the output.

Reply:
@mbasmanova Makes sense, I understand - definitely will be a more reliable approach in that sense. Thanks for the clarification, will make the change.

Reply:
Changes have been made - added a separate method.